### PR TITLE
Backport PR #11420 on branch 3.2.x (Makes ILabShell optional in toc extension)

### PR DESCRIPTION
**Pull request by jweill-aws**
## References

This backports the change from PR #11420 to address issue #11419 in the 3.2.x branch.

## Code changes

See prior PRs

## User-facing changes

See prior PRs

## Backwards-incompatible changes

See prior PRs

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @jweill-aws for doing the manual backport.

Looks like this needs a lint pass.

File: packages/toc-extension/src/index.ts [hunk 1]
```diff
@@ -93,9 +94,11 @@ async function activateTOC(
     label: trans.__('Run Cell(s)')
   });
 
-  app.contextMenu.addItem({
-    selector: '.jp-tocItem',
-    command: CommandIDs.runCells
+  app.commands.addCommand(CommandIDs.showPanel, {
```
Comment (jtpio) [inline packages/toc-extension/src/index.ts:97]: Looks like this command was introduced in https://github.com/jupyterlab/jupyterlab/pull/10425, but not yet backported to 3.2.x.

Comment (jtpio) [reply packages/toc-extension/src/index.ts:97]: So probably we could backport it separately when backporting #10425 for consistency.

**Review by jtpio — APPROVED** [review]
Thanks!

File: packages/toc-extension/src/index.ts [hunk 2]
```diff
@@ -93,11 +93,6 @@ async function activateTOC(
     label: trans.__('Run Cell(s)')
   });
 
-  app.contextMenu.addItem({
-    selector: '.jp-tocItem',
-    command: CommandIDs.runCells
-  });
-
```
Comment (krassowski) [inline packages/toc-extension/src/index.ts:100]: It looks that this PR accidentally removed `runCells` from context menu of ToC.

Comment (krassowski) [reply packages/toc-extension/src/index.ts:100]: Cross-ref https://github.com/jupyterlab/jupyterlab/issues/11584

Comment (krassowski) [reply packages/toc-extension/src/index.ts:100]: Getting fixed in #11617.

Comment (jweill-aws) [reply packages/toc-extension/src/index.ts:100]: @krassowski Sorry about that! Thanks for fixing this.

**Pull request merged**